### PR TITLE
added doorsensor to NukiLock

### PR DESCRIPTION
**Pull request by pree**
I've added the doorsensor state & name to the NukiLock class.
I saw in a different PR that you'd like to have a NukiLockV2 object, since the doorsensor is not present on the V1 Locks.

For this reason I asked in the nuki developer forum what the Bridge-API does with V1-Locks: https://developer.nuki.io/t/door-sensor-state-on-lock-v1/7157

If it just reports as "unknown" or "deactivated" I don't see the needs to split this into two different classes.  What do you think about this?

### CODE REVIEW

**Comment by pschmitt** [comment]
I wasn't aware that it would return "unknown". That alleviates most of my concerns.

File: pynuki/lock.py [hunk 1]
```diff
@@ -10,6 +10,18 @@ class NukiLock(NukiDevice):
     def is_locked(self):
         return self.state == const.STATE_LOCK_LOCKED
 
+    @property
+    def is_door_sensor_activated(self):
+        self.door_sensor_state != const.STATE_DOORSENSOR_DEACTIVATED
```
Comment (pschmitt) [inline pynuki/lock.py:15]: This will return True for v1 locks.
You need to add another check for the "unknown" value.

**Review by pschmitt — CHANGES_REQUESTED** [review]
I'll take a look at what the v1 lock actually returns when I find the time to do so (next week probably - ping me if I forget).

**Comment by pschmitt** [comment]
Okay, so I check with my v1 lock, here's the JSON state:

```python                
{'deviceType': 0,
 'nukiId': REPLACED,
 'name': 'Front Door',
 'firmwareVersion': '1.9.6',
 'mode': 2,
 'state': 1,
 'stateName': 'locked',
 'batteryCritical': False}
```

As you can see there is no door sensor fields. So what we really want to do in `is_door_sensor_activated` is to check for `None` as well.

**Comment by pree** [comment]
Thanks for testing!

I've added Unknown & None to the check. Could you test if this is working for V1, it's working fine for V2

**Comment by pree** [comment]
We've also got an official answer from NUKI that unsupported values are not included in the request. https://developer.nuki.io/t/door-sensor-state-on-lock-v1/7157/2?u=pree

File: pynuki/lock.py [hunk 2]
```diff
@@ -10,6 +10,22 @@ class NukiLock(NukiDevice):
     def is_locked(self):
         return self.state == const.STATE_LOCK_LOCKED
 
+    @property
+    def is_door_sensor_activated(self):
+        return (
+            self.door_sensor_state != const.STATE_DOORSENSOR_DEACTIVATED
+            or self.door_sensor_state != const.STATE_DOORSENSOR_UNKNOWN
+            or self.door_sensor_state != None
+        )
```
Comment (pschmitt) [inline pynuki/lock.py:19]: I've tried your PR with my v1 lock.

In its current state `is_door_sensor_activated` returns `True` for my lock.
I suggest to update the method as follows:

```suggestion
    @property                                                                                  
    def is_door_sensor_activated(self):
        # Nuki v1 locks don't have a door sensor, therefore the                                
        # door_sensor_state will is unset for them.                    
        if (
            not self.door_sensor_state
            or self.door_sensor_state == const.STATE_DOORSENSOR_UNKNOWN    
        ):                                     
            return
        return self.door_sensor_state != const.STATE_DOORSENSOR_DEACTIVATED
```

Comment (pree) [reply pynuki/lock.py]: Perfect, I added your suggestion. When you're ready I would stash my commits and we can merge :)

Comment (pschmitt) [reply pynuki/lock.py]: I squashed the PR. Thanks for your contribution :)

**Comment by pschmitt** [comment]
Included in 1.4.0: https://pypi.org/project/pynuki/1.4.0/

**Comment by chpego** [comment]
Included too Nuki component HA ? :)

**Comment by pschmitt** [comment]
> Included too Nuki component HA ? :)
No. Not yet. I won't be issuing a PR for this.

**Comment by jghaanstra** [comment]
> I won't be issuing a PR for this.
Just wondering why it will not be included in HA?

**Comment by pschmitt** [comment]
> Just wondering why it will not be included in HA?

No reason. Feel free to implement it yourself. I have no way to properly test this, that's all.

**Comment by pree** [comment]
I have a rouge configuration for HA running on my RPi. I will clean it up and create a PR for HA.

**Pull request merged**